java/vector/src/main/java/org/apache/arrow/vector/NullableDecimalVector.java

```diff
@@ -0,0 +1,360 @@
+    * @param buffer   ArrowBuf containing decimal value.
+    */
+   public void setSafe(int index, int start, ArrowBuf buffer) {
+      handleSafe(index);
+      set(index, start, buffer);
+   }
+
+   /**
+    * Same as {@link #set(int, BigDecimal)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param value   BigDecimal containing decimal value.
+    */
+   public void setSafe(int index, BigDecimal value){
+      handleSafe(index);
+      set(index, value);
+   }
+
+   /**
+    * Same as {@link #set(int, NullableDecimalHolder)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param holder  nullable data holder for value of element
+    */
+   public void setSafe(int index, NullableDecimalHolder holder) throws IllegalArgumentException {
+      handleSafe(index);
+      set(index, holder);
+   }
+
+   /**
+    * Same as {@link #set(int, DecimalHolder)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param holder  data holder for value of element
+    */
+   public void setSafe(int index, DecimalHolder holder){
+      handleSafe(index);
+      set(index, holder);
+   }
+
+   /**
+    * Set the element at the given index to null.
+    *
+    * @param index   position of element
+    */
+   public void setNull(int index){
+      handleSafe(index);
+      /* not really needed to set the bit to 0 as long as
+       * the buffer always starts from 0.
+       */
+      BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+   }
+
+   public void set(int index, int isSet, int start, ArrowBuf buffer) {
+      if (isSet > 0) {
+         set(index, start, buffer);
+      } else {
+         BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+      }
+   }
+
+   public void setSafe(int index, int isSet, int start, ArrowBuf buffer) {
+      handleSafe(index);
+      set(index, isSet, start, buffer);
+   }
+
+
+   /******************************************************************
+    *                                                                *
+    *                      vector transfer                           *
+    *                                                                *
+    ******************************************************************/
+
+
+   @Override
+   public TransferPair getTransferPair(String ref, BufferAllocator allocator){
+      return new TransferImpl(ref, allocator);
+   }
+
+   @Override
+   public TransferPair makeTransferPair(ValueVector to) {
+      return new TransferImpl((NullableDecimalVector)to);
+   }
+
+   private class TransferImpl implements TransferPair {
```

Comment:
These TransferImpl class looks very similar in different classes. Can they be refactored out as a single class?